### PR TITLE
Unique names

### PR DESCRIPTION
**Pull request by zxan1285**
- unique names for groups, permissions and roles
- frontend error hangling

### CODE REVIEW

File: server/lib/storage/database.js [hunk 1]
```diff
@@ -54,13 +66,24 @@ export default class Database {
   }
 
   createPermission(permission) {
-    return this.provider
-      .create('permissions', permission);
+    return this.getPermissions()
+      .then(permissions =>
+        checkUnique(
+          permissions.filter(item => item.name === permission.name && item.applicationId === permission.applicationId),
+          `Permission with name "${permission.name}" is already exists.`
```
Comment (sericaia) [inline server/lib/storage/database.js]: this is not an error with a message field, that's why it was not showing the correct error message ;)

File: server/lib/storage/database.js [hunk 2]
```diff
@@ -79,13 +102,24 @@ export default class Database {
...
   }
 
   updateRole(id, role) {
-    return this.provider
-      .update('roles', id, role);
+    return this.getRoles()
+      .then(roles =>
+        checkUnique(
+          roles.filter(item => item.name === role.name && item.applicationId === role.applicationId),
+          `Role with name "${role.name}" is already exists.`,
```
Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Role with name XXX already exists for this application

File: server/lib/storage/database.js [hunk 3]
```diff
@@ -104,13 +138,24 @@ export default class Database {
   }
 
   createGroup(group) {
-    return this.provider
-      .create('groups', group);
+    return this.getGroups()
+      .then(groups =>
+        checkUnique(
+          groups.filter(item => (item.name === group.name)),
+          `Group with name "${group.name}" is already exists.`
```
Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Group with name XXX already exists

File: server/lib/storage/database.js (hunk 1 above)
Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Permission with name XXX already exists for this application

File: server/lib/storage/database.js [hunk 4]
```diff
@@ -54,13 +66,24 @@ export default class Database {
...
   }
 
   updatePermission(id, permission) {
-    return this.provider
-      .update('permissions', id, permission);
+    return this.getPermissions()
+      .then(permissions =>
+        checkUnique(
+          permissions.filter(item => item.name === permission.name && item.applicationId === permission.applicationId),
+          `Permission with name "${permission.name}" is already exists.`,
```
Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Permission with name XXX already exists for this application

**Pull request merged**